Hmac.js

```diff
@@ -48,12 +52,24 @@ export class Hmac {
    *
    * @param {object} options - The options to use.
    * @param {Uint8Array} options.data - The data to sign as a Uint8Array.
+   * @param {boolean} options.useCache - Enable the use of a cache.
    *
    * @returns {Promise<string>} The base64url-encoded signature.
    */
-  async sign({data}) {
+  async sign({data, useCache = true}) {
+    if(useCache) {
+      const sig = this.cache.get(base64url.encode(data));
```

Comment:
Why encode keys?

Reply:
What would you propose? Using a plain object as the key to the cache is not a path to success. The object needs to be transformed into a proper cache key. Is there a simpler way to ensure subsequent equality checks to pull things out of the cache work?